views/js/runner/provider/qti.js

```diff
@@ -0,0 +1,231 @@
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016 (original work) Open Assessment Technologies SA ;
+ */
+/**
+ * Test Runner provider for QTI Tests.
+ *
+ * @author Bertrand Chevrier <bertrand@taotesting.com>
+ */
+define([
+    'jquery',
+    'lodash',
+    'core/promise',
+    'taoTests/runner/areaBroker',
+    'taoTests/runner/proxy',
+    'taoQtiItem/runner/qtiItemRunner',
+    'taoItems/assets/manager',
+    'taoItems/assets/strategies',
+    'tpl!taoQtiTest/runner/provider/layout'
+], function($, _, Promise, areaBroker, proxyFactory, qtiItemRunner, assetManagerFactory, assetStrategies, layoutTpl) {
+    'use strict';
+
+    //states that can be found in the context
+    var states = {
+        initial:       0,
+        interacting:   1,
+        modalFeedback: 2,
+        suspended:     3,
+        closed:        4
+    };
+
+    //the asset strategies
+    var assetManager = assetManagerFactory([
+        assetStrategies.external,
+        assetStrategies.base64,
+        assetStrategies.baseUrl
+    ], { baseUrl: '' });
+
+    /**
+     * A Test runner provider to be registered againt the runner
+     */
+    var qtiProvider = {
+
+        //provider name
+        name : 'qti',
+
+        /**
+         * Initialize and load the area broker with a correct mapping
+         * @returns {areaBroker}
+         */
+        loadAreaBroker : function loadAreaBroker(){
+            var $layout = $(layoutTpl());
+            return areaBroker($layout, {
+                'content' : $('#qti-content', $layout),
+                'toolbox' : $('.tools-box', $layout),
+                'navigation' : $('.navi-box-list', $layout),
+                'control' : $('.top-action-bar .control-box', $layout),
+                'panel' : $('.test-sidebar', $layout),
+                'header' : $('.title-box', $layout)
+            });
+        },
+
+        loadProxy : function loadProxy(){
+
+            var config = this.getConfig();
+
+            var proxyConfig = _.pick(config, [
+                'testDefinition',
+                'testCompilation',
+                'serviceCallId',
+                'serviceController',
+                'serviceExtension'
+            ]);
+            return proxyFactory('qtiServiceProxy', proxyConfig);
+        },
+
+        init : function init(){
+            var self = this;
+
+            var computeNext = function computeNext(action, params){
+
+                var context = self.getTestContext();
+
+                //to be sure load start after unload...
+                //we add an intermediate ns event on unload
+                self.on('unloaditem.' + action, function(){
+                    self.off('.'+action);
+
+                    self.getProxy()
+                        .callItemAction(context.itemUri, action, params)
+                        .then(function(results){
+
+                                self.setTestData(results.testData);
```

Comment:
Maybe I am wrong, but the test data is only got from the init action or the getTestData action, so it will be null after all other actions